notebooks/scripts/consensus_strains.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+from Bio import SeqIO
+from Bio.Align import MultipleSeqAlignment
+from Bio.Align.AlignInfo import SummaryInfo
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+import numpy as np
+import pandas as pd
+from augur.io import read_sequences
+from collections import OrderedDict
+import pandas as pd
+from augur.utils import read_node_data
+import pathlib
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("--alignment", required=True, help="the name of the aligned file to create consensus strains from")
+    parser.add_argument("--metadata", required=True, help="clade information to create cluster per clade.")
+    parser.add_argument("--group-by", default="clade_membership", help="attribute name in metadata to group sequences from the alignment to find a consensus for")
+    parser.add_argument("--output", required=True, help="outputting a csv file of metadata info for HDBSCAN results")
+
+    args = parser.parse_args()
+
+    sequences_by_name = OrderedDict()
+
+    for sequence in read_sequences(args.alignment):#"../ha-na-nextstrain/results/aligned_concatenated.fasta"):
+        sequences_by_name[sequence.id] = str(sequence.seq)
+
+    sequence_names = list(sequences_by_name.keys())
+
+    if ("json" in pathlib.Path(args.metadata).suffix):
+
+        node_data = read_node_data(args.metadata)
+
+        clade_annotations = pd.DataFrame([
+        {"strain": strain, "clade": annotations["MCC"]}
+            for strain, annotations in node_data["nodes"].items()
+            if strain in sequences_by_name
+        ])
+        clade_annotations = clade_annotations.set_index("strain")
+    else:
+        clade_annotations = pd.read_csv(args.metadata, sep="\t").set_index("strain")
+ 
+        if (args.group_by == "tsne_label"):
+            clade_annotations = clade_annotations[["t-sne_label"]]
+        else:
+            clade_annotations = clade_annotations[[args.group_by]]
+        
+        clade_annotations.columns=["clade"]
+
+    clade = clade_annotations.groupby(["clade"])
+    list_clades = [clade.get_group(x) for x in clade.groups]
+
+    # merge strain information (make a dictionary)
+
+    alignments = []
+    for clades in list_clades:
+        sequences = []
+        records = []
+        for index, row in clades.iterrows():
+            sequences.append(sequences_by_name[index])
+            records.append(
+                SeqRecord(Seq(sequences_by_name[index]), id=index)
+            )
+        clades["strain"] = sequences
+
+        alignment = MultipleSeqAlignment(records)
+        consensus_sequence = SummaryInfo(alignment).dumb_consensus(threshold=0.5)
+        alignment.append(SeqRecord(consensus_sequence, id="consensus"))
+
+        cluster_id = clades["clade"].drop_duplicates().values[0]
+        alignments.append(alignment)
```

Comment:
These lines need reorganization:

  1. You don't want to append the consensus to the original alignment; you want to append it to the `alignments` list. Currently, line 74 adds the whole multiple sequence alignment for each cluster to the final alignments output
  2. Line 73 with the cluster id needs to move up a couple of lines and the `cluster_id` needs to be used in the `SeqRecord` instantiation for the consensus above as the `id=` value instead of `"consensus"`. Something like this:

```python
# Get the cluster id as a string, since incoming values can be integers but ids must be strings.
cluster_id = clades["clade"].drop_duplicates().values[0]

# Append consensus sequence to alignment output named by cluster id.
alignments.append(SeqRecord(consensus_sequence, id=cluster_id, name="", description=""))
```

Alternately, this is where you could write the consensus sequence directly to disk instead of building a list of sequences and writing them out later (see the comment below).